### PR TITLE
add parser for mini-notation

### PR DESCRIPTION
**Pull request by neo451**
- Added the parser and corresponding tests for mini-notation
  - should be functionally identical to vortex's MiniVisitor class
  - Pase_mini() returns a lua table representing the parsed ast

- Interpreter would be pretty easy but many pattern functions needed are missing, which I intend to look further into:)

### CODE REVIEW

File: rockspecs/tranquility-dev-1.rockspec [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
 }
 dependencies = {
-    "lua >= 5.1",
-    "luasec >= 1.2.0-1",
-    "losc >= 1.0.1-1",
-    "luasocket >= 3.1.0-1",
-    "abletonlink >= 1.0.0-1"
+	"lua >= 5.1",
+	"luasec >= 1.2.0-1",
+	"losc >= 1.0.1-1",
```
Comment (XiNNiW) [inline rockspecs/tranquility-dev-1.rockspec:18]: Is lpeg built into lua or does it need to go in the dependencies? If its not built in, would you please add lpeg to the lists of dependencies?

File: spec/mini_spec.lua [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	end)
+end)
+
+local function eval(input, query_span, expected_pat)
+	local pat = Mini(input)
+	local span = query_span or TimeSpan:new(0, 1)
+	assert.same(expected_pat, pat.query(span))
+end
+
+-- numbers
```
Comment (XiNNiW) [inline spec/mini_spec.lua]: The test looks great.  Would you please delete the commented `tar` variable at the bottom? If I'm reading it right it seems like it has been replaced by the targets.lua file. (Please push back if I'm reading it wrong!)

File: src/tranquility/mini/targets.lua
Comment (XiNNiW) [inline src/tranquility/mini/targets.lua:1]: Is this file only for organizing expected values for the test? If so, would you please move it to the spec folder?

File: src/tranquility/mini/init.lua [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require("tranquility.mini.visitor")
+
+function Parse_mini(code)
+	local raw_ast = Parse(code)
+	return Visitor:visit(raw_ast)
+end
+
+-- function Mini(code)
```
Comment (XiNNiW) [inline src/tranquility/mini/init.lua]: Would you please delete this commented out version?

File: src/tranquility/mini/visitor.lua [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+		local n_seq = { group }
+		for _, item in pairs(other_seqs) do
+			n_seq[#n_seq + 1] = item
+		end
+		return { type = "random_sequence", elements = n_seq }
+	end
+	return group
+end
+
+-- need a test?
```
Comment (XiNNiW) [inline src/tranquility/mini/visitor.lua]: I'm feeling good with the test you have. It tests the publicly facing functionality (ie. we end up with the correct pattern in the end). I'm fine thinking about exactly how you parsed the AST as an internal implementation detail. Do you agree?

**Review by XiNNiW — COMMENTED** [review]
Thank you for doing this! This looks great. I left some comments with some questions and suggestions. Please feel free to push back!

File: rockspecs/tranquility-dev-1.rockspec (hunk 1 above)
Comment (neo451) [reply rockspecs/tranquility-dev-1.rockspec:18]: yes you r right, I should add it.

File: spec/mini_spec.lua (hunk 2 above)
Comment (neo451) [reply spec/mini_spec.lua]: No, this one is for the interpreter (taken from vortex directly) and have not been tested yet.

Comment (neo451) [reply spec/mini_spec.lua]: maybe should be removed from this commit for now

File: src/tranquility/mini/targets.lua
Comment (neo451) [reply src/tranquility/mini/targets.lua:1]: yes, that is a good idea

File: src/tranquility/mini/init.lua (hunk 3 above)
Comment (neo451) [reply src/tranquility/mini/init.lua]: The idea looks like this: lpeg => a messy, heavily nested AST => Parse_mini() => a meaningful, unified AST => Mini() => the actual pattern. 
And the Mini() can just be plug into the Reify() and have a working mini-notation.

File: src/tranquility/mini/visitor.lua (hunk 4 above)
Comment (neo451) [reply src/tranquility/mini/visitor.lua]: oh yes this is left there for some reason.
About the quality of the test, the targets are from vortex tests, and I also added a few, which is also based on the output of vortex.

**Comment by neo451** [comment]
hi! good morning or afternoon haha, I resolved all the commented fixes and also throw in a try for the ldoc as a documentation tool. I think it is pretty important to have. I tried this the first time today.
you can run `ldoc .` in the main dir to try this out.

File: src/tranquility/mini/init.lua (hunk 3 above)
Comment (XiNNiW) [reply src/tranquility/mini/init.lua]: Ah thanks! I'm starting to see the game plan now. We just need a few more basic functions to allow the AST to be parsed into a pattern. We can leave this here as a guide if that's helpful in the mean time.

**Pull request merged**